src/backend/core/api/viewsets.py

```diff
@@ -696,18 +717,41 @@ def trashbin(self, request, *args, **kwargs):
 
         The selected items are those deleted within the cutoff period defined in the
         settings (see TRASHBIN_CUTOFF_DAYS), before they are considered permanently deleted.
+
+        Optimized version that uses EXISTS instead of expensive subqueries to check
+        owner access on items or their ancestors.
         """
-        queryset = self.queryset.select_related("creator").filter(
-            deleted_at__isnull=False,
-            deleted_at__gte=models.get_trashbin_cutoff(),
+        user = request.user
+
+        # Build the EXISTS subquery to check if user has owner access
+        # to the item or any of its ancestors
+        owner_access_exists = models.ItemAccess.objects.filter(
+            db.Q(user=user) | db.Q(team__in=user.teams),
+            role=models.RoleChoices.OWNER,
+            item__path__ancestors=db.OuterRef("path"),
         )
```

Comment:
Do you think there is a way to avoid code duplication with `annotate_has_owner_role`?